src/PhraseApp.php

```diff
@@ -0,0 +1,251 @@
+            ]);
+
+            if (empty($response)) {
+                $this->client->request('translation.create', [
+                    'project_id' => $this->projectId,
+                    'locale_id' => $this->getLocaleId($message->getLocale()),
+                    'key_id' => $keyId,
+                    'content' => $message->getDomain() . '::' . $message->getTranslation(),
+                ]);
+
+                return;
+            }
+
+            foreach ($response as $translation) {
+                if ($translation['locale']['name'] === $message->getLocale()) {
+                    $id = $translation['id'];
+                }
+                break;
+            }
+
+            if (! isset($id)) {
+                throw new StorageException('Translation id not found.');
+            }
+
+            $this->client->request('translation.update', [
+                'project_id' => $this->projectId,
+                'id' => $id,
+                'content' => $message->getDomain() . '::' . $message->getTranslation(),
+            ]);
+        } catch (\Throwable $e) {
+            throw new StorageException($e->getMessage());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update(Message $message)
+    {
+        $this->create($message);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete($locale, $domain, $key)
+    {
+        try {
+            $response = $this->client->request('key.search', [
+                'project_id' => $this->projectId,
+                'locale_id' => $this->getLocaleId($locale),
+                'q' => $domain . '::' . $key,
+            ]);
+
+            foreach ($response as $keyName) {
+                if ($keyName['name'] === $key) {
+                    $keyId = $key['id'];
+                    break;
+                }
+            }
+
+            if (! isset($keyId)) {
+                return;
+            }
+
+            $this->client->request('key.destroy', [
+                'project_id' => $this->projectId,
+                'id' => $keyId
+
+            ]);
+        } catch (\Throwable $e) {
+            throw new StorageException($e->getMessage());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function export(MessageCatalogueInterface $catalogue)
+    {
+        $locale = $catalogue->getLocale();
+
+        foreach ($catalogue->getDomains() as $domain) {
+            try {
+                $response = $this->client->request('locale.download', [
+                    'project_id' => $this->projectId,
+                    'id' => $this->getLocaleId($locale),
+                    'tag' => $domain,
+                    'file_format' => 'symfony_xliff'
+                ]);
+            } catch (\Throwable $e) {
+                throw new StorageException($e->getMessage());
+            }
+
+            $data = $response['text'];
+            /* @var \GuzzleHttp\Stream\Stream $data */
```

Comment:
I think it is more common to move this comment above `$data = $response['text'];`